src/OpenRiaServices.Hosting.Wcf/Framework/WCF/Behaviors/DomainOperationInvoker.cs

```diff
@@ -134,5 +142,241 @@ private DomainService GetDomainService(object instance, WcfDomainServiceContext
+                        string[] dependencies = outputCacheInfo.SqlCacheDependencies.Split(semiColonDelimiter, StringSplitOptions.RemoveEmptyEntries);
+                        foreach (string dependency in dependencies)
+                        {
+                            string[] dependencyTokens = dependency.Split(colonDelimiter, StringSplitOptions.RemoveEmptyEntries);
+                            if (dependencyTokens.Length != 2)
+                            {
+                                throw new InvalidOperationException(Resource.DomainService_InvalidSqlDependencyFormat);
+                            }
+
+                            context.Response.AddCacheDependency(new SqlCacheDependency(dependencyTokens[0], dependencyTokens[1]));
+                        }
+                    }
+
+                    if (!String.IsNullOrEmpty(outputCacheInfo.VaryByHeaders))
+                    {
+                        string[] headers = outputCacheInfo.VaryByHeaders.Split(semiColonDelimiter, StringSplitOptions.RemoveEmptyEntries);
+                        foreach (string header in headers)
+                        {
+                            policy.VaryByHeaders[header] = true;
+                        }
+                    }
+
+                    // The cache is based on the values of the domain operation entry's parameters.
+                    foreach (DomainOperationParameter pi in domainOperationEntry.Parameters)
+                    {
+                        policy.VaryByParams[pi.Name] = true;
+                    }
+
+                    // We don't cache when query parameters are used. We need to vary by query parameters 
+                    // though such that we can intercept requests with query parameters and by-pass the cache.
+                    foreach (string queryParameter in supportedQueryParameters)
+                    {
+                        policy.VaryByParams[queryParameter] = true;
+                    }
+
+                    return;
+                }
+            }
+
+            // By default, don't let clients/proxies cache anything.
+            context.Response.Cache.SetCacheability(HttpCacheability.NoCache);
+        }
+
+        /// <summary>
+        /// Determines whether caching is supported for the current request to the specified domain operation entry.
+        /// </summary>
+        /// <param name="context">The context for the request.</param>
+        /// <param name="domainOperationEntry">The requested domain operation entry, if any.</param>
+        /// <returns>True if caching is supported.</returns>
+        private static bool SupportsCaching(HttpContext context, DomainOperationEntry domainOperationEntry)
+        {
+            if (domainOperationEntry != null
+                && context.Request.HttpMethod.Equals("GET", StringComparison.OrdinalIgnoreCase))
+            {
+                foreach (string queryParameter in supportedQueryParameters)
```

Comment:
I understand that the code is only moved, bug maybe we should change the name to something like "s_**un**supportedQueryParameters" based on the logit I beleive having any of those parameters present will disable caching ?

Reply:
I read the code and i think you are right, the name seems wrong ;-) Fixed :-)